### DISCUSSION TITLE
(iOS) Action failure due to location

### DISCUSSION THREAD

**Discussion by arashnrim** [community/community#25943] [Actions]
<p>Hello there! Apologies for the ambiguity of the topic title.</p>
<p>I’m having an issue that prevents me from running my workflow properly. Here’s <a href="https://github.com/arashnrim/Upgrade/tree/add-workflow" rel="nofollow noopener">a link to my repository</a> to help.</p>
<p>All my previous attempts have failed so far, and I’m suspecting that the issue at hand is the location of the files. Referring to my repository, the structure of my repository is as such:</p>
<pre><code class="lang-auto">Upgrade
  Podfile
  Upgrade
  Upgrade.xcworkspace
  Upgrade.xcproj
  Pods
  Podfile.lock
</code></pre>
<p>In my action, I wanted to run Cocoapods to install all the dependencies for the project through <code>pod install</code>. However, when I did <code>cd Upgrade/</code> the action fails and returns:</p>
<pre><code class="lang-auto">line 1: cd: Upgrade/: No such file or directory
</code></pre>
<p>What should I do to resolve this problem? All help is appreciated, thank you!</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/arashnrim">@arashnrim</a>,</p>
<p>In a workflow, if some steps need to read or update the source files in the repository, the runner won’t automatically check out the source files or directly access the files from the remote GitHub repository.</p>
<p>Before the steps which need to access the source files, you should setup a step to check out the source files into the workspace (<strong>github.workspace</strong>) on the runner machine. You can use the <a href="https://github.com/actions/checkout#checkout-v2" rel="nofollow noopener"><strong>Checkout</strong> action</a> to do this.</p>
<pre><code class="lang-yaml">- name: Checkout
  uses: actions/checkout@v2
  with:
    . . .
</code></pre>
<p>I checked the workflow file you used, but did not find any step was used to check out the source files (see <a href="https://github.com/arashnrim/Upgrade/blob/ba51bd1edabc6dec411df32b353e2544f2db1ab8/.github/workflows/build.yml" rel="nofollow noopener">here</a>). No source files were checked out into the workspace, so the message “<strong>No such file or directory</strong>” was returned.<br>
You can change your workflow like as below:</p>
<pre><code class="lang-yaml">
name: Build Xcode project

on: [push, pull_request]

jobs:
  build:
    runs-on: macos-latest
    steps:
      - name: Install Cocoapods
        run: |
          sudo gem install cocoapods

# checkout the source files from the remote GitHub repository
# so that the subsequent steps can access the source files
      - name: Checkout
        uses: actions/checkout@v2

      - name: Install dependencies
        run: |
          cd Upgrade/
          pod install

      - name: Build project
        run: |
          xcodebuild -workspace "Upgrade.xcworkspace" -scheme "Upgrade"
</code></pre>

**Comment by arashnrim** [comment]
<p>Awesome, that did the trick! Thanks a lot!</p>
